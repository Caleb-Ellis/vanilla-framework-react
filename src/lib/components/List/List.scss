@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_lists';
@include vf-b-typography;
@include vf-p-lists;
