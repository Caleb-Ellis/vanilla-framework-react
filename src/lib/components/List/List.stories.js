import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import List from './List';
import ListItem from './ListItem';

storiesOf('List', module)
  .add('Default',
    withInfo('If you want to display lists in a way that is more visually distinctive than the standard <ol> and <ul>, Vanilla has 7 list styles at your disposal.')(() => (
      <List>
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>),
    ),
  )

  .add('Ticked',
    withInfo('Add the prop "ticked" to each list item to display tick icons.')(() => (
      <List>
        <ListItem ticked>Lorem</ListItem>
        <ListItem ticked>Ipsum</ListItem>
        <ListItem ticked>Dolor</ListItem>
      </List>),
    ),
  )

  .add('Divided',
    withInfo('Use the prop "divided" to add horizontal lines between the items.')(() => (
      <List modifier="divided">
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>),
    ),
  )

  .add('Ticked and Divided',
    withInfo('You can combine both "ticked" and "divided" to style a list with horizontal dividers and tick icons.')(() => (
      <List modifier="divided">
        <ListItem ticked>Lorem</ListItem>
        <ListItem ticked>Ipsum</ListItem>
        <ListItem ticked>Dolor</ListItem>
      </List>),
    ),
  )

  .add('Inline List',
    withInfo('Apply the modifier "inline" to display all the list items in one line.')(() => (
      <List modifier="inline">
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>),
    ),
  )

  .add('Middot-separated Inline List',
    withInfo('Apply the modifier "middot" to add a middot character between inline list items.')(() => (
      <List modifier="middot">
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>),
    ),
  )

  .add('Split List',
    withInfo('If you wish to split the items in a list into two columns above $breakpoint-medium, you can do so by adding the modifier "split" to the List component.')(() => (
      <List modifier="split">
        <ListItem ticked>Lorem</ListItem>
        <ListItem ticked>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean nec ipsum augue. Ut arcu
          erat, lacinia sit ametjusto quis. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </ListItem>
        <ListItem ticked>Dolor</ListItem>
        <ListItem ticked>
          Aenean nec ipsum augue. Ut arcu erat, lacinia sit amet justo quis.
        </ListItem>
        <ListItem ticked>Ipsum</ListItem>
        <ListItem ticked>Ut arcu erat, lacinia sit amet justo quis.</ListItem>
      </List>),
    ),
  );
