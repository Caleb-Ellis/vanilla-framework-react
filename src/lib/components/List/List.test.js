import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import List from './List';
import ListItem from './ListItem';

describe('List component', () => {
  it('should render a simple list correctly', () => {
    const list = ReactTestRenderer.create(
      <List>
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a ticked list correctly', () => {
    const list = ReactTestRenderer.create(
      <List>
        <ListItem ticked>Lorem</ListItem>
        <ListItem ticked>Ipsum</ListItem>
        <ListItem ticked>Dolor</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a divided list correctly', () => {
    const list = ReactTestRenderer.create(
      <List modifier="divided">
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a ticked and divided list correctly', () => {
    const list = ReactTestRenderer.create(
      <List modifier="divided">
        <ListItem ticked>Lorem</ListItem>
        <ListItem ticked>Ipsum</ListItem>
        <ListItem ticked>Dolor</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an inline list correctly', () => {
    const list = ReactTestRenderer.create(
      <List modifier="inline">
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a middot list correctly', () => {
    const list = ReactTestRenderer.create(
      <List modifier="middot">
        <ListItem>Lorem</ListItem>
        <ListItem>Ipsum</ListItem>
        <ListItem>Dolor</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a split list correctly', () => {
    const list = ReactTestRenderer.create(
      <List modifier="split">
        <ListItem ticked>Lorem</ListItem>
        <ListItem ticked>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean nec ipsum augue. Ut arcu
          erat, lacinia sit ametjusto quis. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </ListItem>
        <ListItem ticked>Dolor</ListItem>
        <ListItem ticked>
          Aenean nec ipsum augue. Ut arcu erat, lacinia sit amet justo quis.
        </ListItem>
        <ListItem ticked>Ipsum</ListItem>
        <ListItem ticked>Ut arcu erat, lacinia sit amet justo quis.</ListItem>
      </List>,
    );
    const json = list.toJSON();
    expect(json).toMatchSnapshot();
  });
});
