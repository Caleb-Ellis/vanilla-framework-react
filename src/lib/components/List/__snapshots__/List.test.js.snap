// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render a divided list correctly 1`] = `
<ul
  className="p-list--divided"
>
  <li
    className="p-list__item"
  >
    Lorem
  </li>
  <li
    className="p-list__item"
  >
    Ipsum
  </li>
  <li
    className="p-list__item"
  >
    Dolor
  </li>
</ul>
`;

exports[`List component should render a middot list correctly 1`] = `
<ul
  className="p-inline-list--middot"
>
  <li
    className="p-inline-list__item"
  >
    Lorem
  </li>
  <li
    className="p-inline-list__item"
  >
    Ipsum
  </li>
  <li
    className="p-inline-list__item"
  >
    Dolor
  </li>
</ul>
`;

exports[`List component should render a simple list correctly 1`] = `
<ul
  className="p-list"
>
  <li
    className="p-list__item"
  >
    Lorem
  </li>
  <li
    className="p-list__item"
  >
    Ipsum
  </li>
  <li
    className="p-list__item"
  >
    Dolor
  </li>
</ul>
`;

exports[`List component should render a split list correctly 1`] = `
<ul
  className="p-list--divided is-split"
>
  <li
    className="p-list__item is-ticked"
  >
    Lorem
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean nec ipsum augue. Ut arcu erat, lacinia sit ametjusto quis. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Dolor
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Aenean nec ipsum augue. Ut arcu erat, lacinia sit amet justo quis.
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Ipsum
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Ut arcu erat, lacinia sit amet justo quis.
  </li>
</ul>
`;

exports[`List component should render a ticked and divided list correctly 1`] = `
<ul
  className="p-list--divided"
>
  <li
    className="p-list__item is-ticked"
  >
    Lorem
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Ipsum
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Dolor
  </li>
</ul>
`;

exports[`List component should render a ticked list correctly 1`] = `
<ul
  className="p-list"
>
  <li
    className="p-list__item is-ticked"
  >
    Lorem
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Ipsum
  </li>
  <li
    className="p-list__item is-ticked"
  >
    Dolor
  </li>
</ul>
`;

exports[`List component should render an inline list correctly 1`] = `
<ul
  className="p-inline-list"
>
  <li
    className="p-inline-list__item"
  >
    Lorem
  </li>
  <li
    className="p-inline-list__item"
  >
    Ipsum
  </li>
  <li
    className="p-inline-list__item"
  >
    Dolor
  </li>
</ul>
`;
